src/Morphir/JsonSchema/AST.elm

```diff
@@ -8,18 +8,54 @@ type alias TypeName =
 
 
 type alias Schema =
-    { dirPath : List String
-    , fileName : String
-    , id : String
+    { id : String
     , schemaVersion : String
     , definitions : Dict TypeName SchemaType
     }
 
 
+
+-- Used to enforce a Set type
+
+
+type alias UniqueItems =
+    Bool
+
+
+
+{-
+   The length of the array can be specified using the minItems and maxItems keywords.
+   The numberOfItems is used to enforce this
+   |
+-}
+
+
+type alias NumberOfItems =
+    Int
+
+
 type SchemaType
     = Integer
-    | Array SchemaType
-    | String
+    | Array ArrayType UniqueItems
+    | String Derivative
     | Number
     | Boolean
     | Object (Dict String SchemaType)
+    | Const String
+    | Ref TypeName
+    | OneOf (List SchemaType)
+    | Null
+
+
+type ArrayType
+    = ListType SchemaType
+    | TupleType (List SchemaType) NumberOfItems
+
+
+type Derivative
+    = BasicString
+    | CharString
+    | DecimalString
+    | DateString
+    | TimeString
+    | MonthString
```

Comment:
I'm not sure if this is a good idea. This structure is very far from how JSON Schema represents this. Feels like it should refer to format in the name. I'm also not sure about having a concept of Decimal or Month here. This is the JSON Schema AST and that one doesn't include these concepts. Maybe if you do something like this:

```elm
    | String StringConstraints

type alias StringConstraints =
    { format : Maybe String
    }

```